src/worker.cpp

```diff
@@ -1563,4 +1563,100 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    if (i_dbusObjPath.empty())
+    {
+        throw std::runtime_error("Given DBus object path is empty.");
+    }
+
+    const std::string& l_fruPath =
+        jsonUtility::getFruPathFromJson(m_parsedJson, i_dbusObjPath);
+
+    try
+    {
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            throw std::runtime_error("Given FRU is not present");
+        }
+        else
+        {
+            if (jsonUtility::isActionRequired(m_parsedJson, l_fruPath,
+                                              "preAction", "deletion"))
+            {
+                if (!jsonUtility::executeBaseAction(m_parsedJson, "preAction",
```

Comment:
why worker functions for processPreAction not used ?

Reply:
Jinu's point is valid. Please revert it back to processPreAction API call.

Reply:
Done